Assets/MixedRealityToolkit.Staging/UnityAR/Editor/ConfigurationChecker/UnityARConfigurationChecker.cs

```diff
@@ -15,78 +17,108 @@ public class UnityARConfigurationChecker
-                return null;
+                ScriptingUtilities.AppendScriptingDefinitions(BuildTargetGroup.Android, definitions);
+                ScriptingUtilities.AppendScriptingDefinitions(BuildTargetGroup.iOS, definitions);
+                return true;
+            }
+            else
+            {
+                ScriptingUtilities.RemoveScriptingDefinitions(BuildTargetGroup.Android, definitions);
+                ScriptingUtilities.RemoveScriptingDefinitions(BuildTargetGroup.iOS, definitions);
+                return false;
             }
-
-            return files[0];
         }
 
         /// <summary>
-        /// Determines if a file update is needed and returns the replacement.
+        /// Updates the assembly definition to contain the appropriate references based on the Unity
+        /// version and if the project contains the AR Foundation package.
         /// </summary>
-        /// <param name="file">The reference file used to determine the new file.</param>
-        /// <returns>FileInfo object representing the file, or null if a new file is not needed.</returns>
-        private static FileInfo GetNewAsmDefFile(FileInfo file)
+        private static void UpdateAsmDef(bool arFoundationPresent)
         {
-            // Read the file.
-            string fileContents = string.Empty;
-            using (FileStream fs = new FileStream(file.FullName, FileMode.Open, FileAccess.Read))
+            string asmDefFileName = "Microsoft.MixedReality.Toolkit.Providers.UnityAR.asmdef";
+            FileInfo[] asmDefFiles = FileUtilities.FindFilesInAssets(asmDefFileName);
+
+            if (asmDefFiles.Length == 0)
             {
-                using (StreamReader reader = new StreamReader(fs))
-                {
-                    fileContents = reader.ReadToEnd();
-                }
+                Debug.LogWarning($"Unable to locate file: {asmDefFileName}");
+                return;
+            }
+
+            AssemblyDefinition asmDef = AssemblyDefinition.Load(asmDefFiles[0].FullName);
+            if (asmDef == null)
+            {
+                Debug.LogWarning($"Unable to load file: {asmDefFileName}");
+                return;
+            }
+
+            List<string> references = new List<string>();
+            if (asmDef.References != null)
+            {
+                references.AddRange(asmDef.References);
             }
 
-            // Key values used to determine if a replacement is required.
+            // Assemblies required by Unity 2018 and / or 2019 (and newer)
             const string arFoundationReference = "Unity.XR.ARFoundation";
             const string spatialTrackingReference = "UnityEngine.SpatialTracking";
+            bool changed = false;
 
-            string updateTo = string.Empty;
-#if UNITY_2018
-            if (!fileContents.Contains(arFoundationReference) ||
-                fileContents.Contains(spatialTrackingReference))
+            if (arFoundationPresent)
```

Comment:
>arFoundationPresent [](start = 16, length = 19)

Shouldn't this entire function be wrapped in if (arFoundationPresent)?

It looks like the only part that does stuff in this function (i.e. modifying the asmdef) is in this check, and none of the stuff above is useful to do if arFoundationPresent is false.